x/mongo/driver/topology/rtt_monitor.go

```diff
@@ -206,7 +207,7 @@ func percentile(perc float64, samples []time.Duration, minSamples int) time.Dura
 			floatSamples = append(floatSamples, float64(sample))
 		}
 	}
-	if len(floatSamples) < minSamples {
+	if len(floatSamples) == 0 || len(floatSamples) < minSamples {
```

Comment:
We also want to return `0` when there are no viable samples in `samples`, as we do for the `min` function.